fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -425,23 +484,29 @@ Optional<ACLHandle> getEffectiveAcl(final FedoraResource resource, final boolean
     }
 
     private List<WebACAuthorization> getDefaultAuthorizations() {
-        final Map<String, List<String>> aclTriples = new HashMap<>();
         final List<WebACAuthorization> authorizations = new ArrayList<>();
 
-        getDefaultAcl(null, authPropsConfig.getRootAuthAclPath())
-                .listStatements().mapWith(Statement::asTriple).forEachRemaining(triple -> {
-            if (hasAclPredicate.test(triple)) {
-                final String predicate = triple.getPredicate().getURI();
-                final List<String> values = aclTriples.computeIfAbsent(predicate,
-                    key -> new ArrayList<>());
-                nodeToStringStream(triple.getObject()).forEach(values::add);
-                if (predicate.equals(WEBAC_AGENT_VALUE)) {
-                    additionalAgentValues(triple.getObject()).forEach(values::add);
+        final var defaultAcls = getDefaultAcl(null, authPropsConfig.getRootAuthAclPath());
+        final var aclSubjects = defaultAcls.listSubjects();
+
+        aclSubjects.forEach(aclResource -> {
+            final Map<String, List<String>> aclTriples = new HashMap<>();
+            aclResource.listProperties().mapWith(Statement::asTriple).forEach(aclTriple -> {
+                if (hasAclPredicate.test(aclTriple)) {
```

Comment:
Just wondering if there is a reason you didn't add this as a `filter(hasAclPredicate)` between the `.mapWith(Statement::asTriple)` and `forEach(aclTriple ->..` ? 
Not sure it has any impact on performance and there might been a reason that I'm missing.

Reply:
Looking back, this is similar to how the logic was before (with the `hashAclPredicate.test` separate). I'm not really sure why, my only thought is that this is using iterators from jena rather than a stream so it's possible filter wasn't available when this was originally done. That said jena provides `filterDrop` and `filterKeep` so I'll see about using one of those.

Reply:
Don't worry about it, I really just wanted to know if there was a reason I wasn't seeing. If we find a delay we can review later.